### PR TITLE
Feature/ansi codecs

### PR DESCRIPTION
**Pull request by omerbenamram**
Allows the parser to control the ANSI codepage used in inner single-byte encoded strings.

### CODE REVIEW

File: src/bin/evtx_dump.rs [hunk 1]
```diff
@@ -174,6 +182,16 @@ fn main() {
...
         )
+        .arg(
+            Arg::with_name("ansi-codec")
+                .long("--ansi-codec")
+                .possible_values(&encodings().iter()
+                    .filter(|&e| e.raw_decoder().is_ascii_compatible())
+                    .map(|e| e.name())
+                    .collect::<Vec<&'static str>>())
+                .default_value(encoding::all::WINDOWS_1252.name())
+                .help("When set, controls the codec of ansi encoded strings the file. defaults to `WINDOWS-1252`"),
```
Comment (ohadravid) [inline src/bin/evtx_dump.rs]: Put a link to the docs where all the encodings are defined?

File: src/binxml/value_variant.rs [hunk 2]
```diff
@@ -207,170 +211,159 @@ impl<'a> BinXmlValue<'a> {
...
+        trace!("deserialize_value_type: {:?}, {:?}", value_type, size);
+
+        let value = match (value_type, size) {
+            (BinXmlValueType::NullType, _) => BinXmlValue::NullType,
+            (BinXmlValueType::StringType, Some(sz)) => BinXmlValue::StringType(Cow::Owned(
+                read_utf16_by_size(cursor, u64::from(sz))
+                    .context(err::FailedToDecodeUTF16String {
+                        offset: cursor.position(),
+                    })?
+                    .unwrap_or_else(|| "".to_owned()),
```
Comment (ohadravid) [inline src/binxml/value_variant.rs]: Shouldn't this be an error and not a default fallback?

File: src/bin/evtx_dump.rs (hunk 1 above)
Comment (omerbenamram) [reply src/bin/evtx_dump.rs]: It prints it nicely if you use `evtx_dump --help`

File: src/binxml/value_variant.rs (hunk 2 above)
Comment (omerbenamram) [reply src/binxml/value_variant.rs]: Notice that fallback is on the `Option<String>` part of the `Result<Option<String>>`
Strings are allowed to be empty, but we don't accept I/O errors.

**Pull request merged**